js/util/parse.test.js

```diff
@@ -0,0 +1,135 @@
+// Tests for parsing algorithms from js/util/util.js
+import { parseAnd, parseOr, parseCourse } from "./util"
+
+const recursiveChecker = (actual, expected) => {
```

Comment:
This is good, but I don't think it's necessary. You should try using Jest's `equals` function: https://jestjs.io/docs/expect#thisequalsa-b.